### PR TITLE
Networking

### PR DESCRIPTION
**Pull request by MatthewCHill**

### CODE REVIEW

File: ConnectionJustForToday/ConnectionJustForToday/HelperFiles/JFTService.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
+//
+//  JFTService.swift
+//  ConnectionJustForToday
+//
+//  Created by Matthew Hill on 4/4/23.
+//
+
+import Foundation
+import Collections
```
Comment (maxpoff) [inline ConnectionJustForToday/ConnectionJustForToday/HelperFiles/JFTService.swift]: if you're no longer using an ordered dictionary, you do not need to import collections (which, I agree that you should not be using an ordered dictionary)

File: ConnectionJustForToday/ConnectionJustForToday/Models/JFTReading.swift [hunk 2]
```diff
@@ -6,30 +6,67 @@
 //
 
 import Foundation
+import CryptoKit
```
Comment (maxpoff) [inline ConnectionJustForToday/ConnectionJustForToday/Models/JFTReading.swift]: Where are you using CryptoKit?

File: ConnectionJustForToday/ConnectionJustForToday/Models/JFTReading.swift [hunk 3]
```diff
@@ -6,30 +6,67 @@
...
+    var reference: String
     var body: String
-    var afffirmation: String
+    var affirmation: String
+    var copyright: String
     var postsJFT: [JFTPost]?
     var uuid: String
     
-    init(date: String, title: String, quote: String, body: String, afffirmation: String, posts: [JFTPost] = [], uuid: String = UUID().uuidString) {
+    init(date: String, title: String, pageNumber: String, quote: String, reference: String, body: String, affirmation: String, copyright: String, posts: [JFTPost] = [], uuid: String = UUID().uuidString) {
```
Comment (maxpoff) [inline ConnectionJustForToday/ConnectionJustForToday/Models/JFTReading.swift:39]: I think this UUID is going to cause the problem that we talked about, where posts are not visible across instances of the reading

File: ConnectionJustForToday/ConnectionJustForToday/Views/CreateUser/CreateUserViewController.swift [hunk 4]
```diff
@@ -9,13 +9,26 @@ import UIKit
...
+// MARK: - Actions
+    @IBAction func createUserButtonTapped(_ sender: Any) {
+        guard let email = userEmailTextField.text,
+              let password = userPasswordTextField.text,
+              let confrimPassword = confirmPasswordTextField.text else { return }
+        
+        viewModel.createUser(email: email, password: password, confrimPassword: confrimPassword)
+    }
     /*
     // MARK: - Navigation
```
Comment (maxpoff) [inline ConnectionJustForToday/ConnectionJustForToday/Views/CreateUser/CreateUserViewController.swift]: If this is unused, let's remove the unnecessary comments

File: ConnectionJustForToday/ConnectionJustForToday/Views/CreateUser/CreateUserViewModel.swift [hunk 5]
```diff
@@ -6,7 +6,46 @@
...
+    func signOut() {
+        let firebaseAuth = Auth.auth()
+        do {
+            try firebaseAuth.signOut()
+            print("Signed Out")
+        } catch let signOutError as NSError {
+            print("Error signing out", signOutError)
+        }
+    }
+
```
Comment (maxpoff) [inline ConnectionJustForToday/ConnectionJustForToday/Views/CreateUser/CreateUserViewModel.swift]: extra white space

File: ConnectionJustForToday/ConnectionJustForToday/Views/JFTReading/JFTReadingViewController.swift [hunk 6]
```diff
@@ -7,12 +7,56 @@
 
 import UIKit
 
-class JFTReadingViewController: UIViewController {
-
+class JFTReadingViewController: UIViewController{
+    
+    
+    // MARK: - Outlets
+
```
Comment (maxpoff) [inline ConnectionJustForToday/ConnectionJustForToday/Views/JFTReading/JFTReadingViewController.swift:13]: extra white space

File: ConnectionJustForToday/ConnectionJustForToday/Views/JFTReading/JFTReadingViewController.swift [hunk 7]
```diff
@@ -7,12 +7,56 @@
...
-
+        viewModel = JFTReadingViewModel(delegate: self)
     }
-
     
+    // MARK: - Properties
+    var viewModel: JFTReadingViewModel!
+    
+    // MARK: - Functions
+
```
Comment (maxpoff) [inline ConnectionJustForToday/ConnectionJustForToday/Views/JFTReading/JFTReadingViewController.swift:32]: Extra white space. Remove all extra white space beneath your marks. I won't comment on the rest of them

File: ConnectionJustForToday/ConnectionJustForToday/Views/SPADReading/SPADReadingViewController.swift [hunk 8]
```diff
@@ -7,11 +7,48 @@
...
+} // End of class
 
+// MARK: - Extension
+
+extension SPADReadingViewController: SpadReadingViewModelDelegate {
+    func scrapedReadingSuccessful() {
+        DispatchQueue.main.async {
+            self.updateUI()
+        }
     }
```
Comment (maxpoff) [inline ConnectionJustForToday/ConnectionJustForToday/Views/SPADReading/SPADReadingViewController.swift]: white space. Clean up all your closing bracket white spaces

File: ConnectionJustForToday/ConnectionJustForToday/Models/JFTReading.swift (hunk 2 above)
Comment (MatthewCHill) [reply ConnectionJustForToday/ConnectionJustForToday/Models/JFTReading.swift]: Removed. I added when we were talking about a way for the posts to be dynamic.

File: ConnectionJustForToday/ConnectionJustForToday/Views/SPADReading/SPADReadingViewController.swift (hunk 8 above)
Comment (MatthewCHill) [reply ConnectionJustForToday/ConnectionJustForToday/Views/SPADReading/SPADReadingViewController.swift]: Done.

File: ConnectionJustForToday/ConnectionJustForToday/Views/CreateUser/CreateUserViewController.swift (hunk 4 above)
Comment (MatthewCHill) [reply ConnectionJustForToday/ConnectionJustForToday/Views/CreateUser/CreateUserViewController.swift]: fixed

File: ConnectionJustForToday/ConnectionJustForToday/Views/JFTReading/JFTReadingViewController.swift (hunk 6 above)
Comment (MatthewCHill) [reply ConnectionJustForToday/ConnectionJustForToday/Views/JFTReading/JFTReadingViewController.swift:13]: fixed

File: ConnectionJustForToday/ConnectionJustForToday/Models/JFTReading.swift (hunk 3 above)
Comment (MatthewCHill) [reply ConnectionJustForToday/ConnectionJustForToday/Models/JFTReading.swift:39]: Is it okay to leave for now and work on the SHA Tomorrow?

**Review by maxpoff — APPROVED** [review]

**Pull request merged**